DEVEL.rst

```diff
@@ -8,132 +8,176 @@ If something does not work as expected, please create an issue or
-sudo dnf install -y etcd redis
-sudo dnf install -y @development-tools redhat-rpm-config
-sudo dnf install -y python3 python3-virtualenv
-```
+First install the required packages:
+
+.. code-block:: shell
+
+   $ sudo dnf update -y
+   $ sudo dnf install -y etcd redis
+   $ sudo dnf install -y @development-tools redhat-rpm-config
+   $ sudo dnf install -y python3 python3-virtualenv
+
+
+Next enable the ``etcd`` and ``redis`` services:
+
+.. code-block:: shell
+
+   $ systemctl enable etcd redis
+   $ systemctl start etcd redis
 
-```
-$ systemctl enable etcd redis
-$ systemctl start etcd redis
-```
 
 
-## Pull repositories
+Pull repositories
+-----------------
+
 Commissaire consists of several projects that either provide
-common code or the actual services.
+common code or the actual services. You can clone the repos
+by following the example below:
+
+.. code-block:: shell
+
+   $ mkdir commissaire-projects
+   $ cd commissaire-projects
+   $ git clone https://github.com/projectatomic/commissaire
+   $ git clone https://github.com/projectatomic/commissaire-service
+   $ git clone https://github.com/projectatomic/commissaire-http
+   $ git clone https://github.com/projectatomic/commctl
 
-```
-mkdir commissaire-projects
-cd commissaire-projects
-git clone https://github.com/projectatomic/commissaire
-git clone https://github.com/projectatomic/commissaire-service
-git clone https://github.com/projectatomic/commissaire-http
-git clone https://github.com/projectatomic/commctl
-```
 
 After this we will install each project to setup a development
 environment.
 
 
-### Create VirtualEnv
-```
-virtualenv-3.5 devel
-. devel/bin/activate
-```
+Create VirtualEnv
+-----------------
+
+.. code-block::
+
+   $ virtualenv-3.5 devel
+   $ . devel/bin/activate
+
 
 Continue executing the following commands in the virtualenv you just created.
 
 
-## Install Commissaire
-```
-cd commissaire
-pip install -e .
-popd
-```
+Install Commissaire
+===================
 
+.. code-block::
 
-### Install Commissaire Service
-```
-cd commissaire-service
-pip install -e .
-tools/etcd_init.sh
-```
+   $ cd commissaire
+   $ pip install -e .
+   $ popd
 
-Edit your storage configuration to point to your etcd instance
-```
-cp conf/storage.conf mystorage.conf
-```
-Note: point server_url to http://127.0.0.1:2379 (not https)
+
+Install Commissaire Service
+---------------------------
+
+.. code-block::
+
+   $ cd commissaire-service
+   $ pip install -e .
+   $ tools/etcd_init.sh
+
+
+Edit your storage configuration to point to your etcd instance:
+
+.. code-block::
+
+   $ cp conf/storage.conf mystorage.conf
+
+
+**Note**: Point ``server_url`` to ``http://127.0.0.1:2379`` (not https)
 
 
 Start the service
-```
-commissaire-storage-service -c mystorage.conf &
-popd
-```
 
-### Install Commissaire Server
-```
-cd commissaire-http
-pip install -e .  # Install commissaire-http into the virtualenv
-```
+.. code-block::
+
+   $ commissaire-storage-service -c mystorage.conf &
```

Comment:
This is a bug. It's complaining that ``bus-uri`` is not set and it looks like we accidentally removed that from the configuration file parsing. The workaround for now is:

```shell
commissaire-storage-service -c mystorage.conf --bus-uri redis://127.0.0.1:6379 &
```

/cc @mbarnes